### DISCUSSION TITLE
Problem with timeseries data replication between nodes

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#17244] [General]
Hi,

I have a collection upon which we're collecting timeseries information for the docs. The data collected can change when further actions are taken related to the document so we have a process where we delete a subset of the ts data and recreated it with new entries based on a re-evaluated calculation.

Confirmed have unique timestamps for each entry.

I am frequently seeing the ts data disappearing, sometimes completely, sometimes partially and am pretty sure this is related to replication between nodes (I have not seem this happen at all in our test environments which are single node clusters)

I am also seeing a different number of data points on each node after running some code to sync the data correctly, see screenshots (14 on node B and C and 16 data points on node A

Node A
<img width="568" alt="image" src="https://github.com/ravendb/ravendb/assets/3949005/84cfdc72-3e32-4ba5-bd87-f8976cd6d1f5">
Node B
<img width="654" alt="image" src="https://github.com/ravendb/ravendb/assets/3949005/e9685e09-d9f6-4fc2-9f21-0dfc801d35d3">
Node C
<img width="807" alt="image" src="https://github.com/ravendb/ravendb/assets/3949005/3bb51d84-bb21-4bce-bca6-508db5d78738">

After a new seconds Node A has 2 entries deleted, so its in sync with the other nodes, but there should be 16 entries.

So it looks like Node A is being updated with the correct # of entries, but its getting overwritten by Node B or Node C which has the incorrect # of entries.

Any idea what might be going on here?

Thanks

**Comment by shiranshalom** [comment]
Hi Nick,

Distributed clients and nodes can modify time series concurrently; the modifications are merged by the cluster without conflict (please see https://ravendb.net/docs/article-page/5.4/csharp/document-extensions/timeseries/design#no-conflicts for more details). Therefore, by design, we prefer the higher value (if the entry exists in storage AND the incoming from replicated segment both have values) OR the deleted value between the two. 
Please note that the above happens when there are existing entries/segments between the range in storage. 
You can bypass this by confirming all nodes are in sync before updating the values - this will allow you to update the entire segment directly without comparing the actual values. BUT, it is important to note that this is not guaranteed due to other potential writes to the DB. 

If you have more questions feel free to contact us again.
Thanks
